quickstart/101-managed-instance/main.tf

```diff
@@ -0,0 +1,85 @@
+# Create resource group
+resource "azurerm_resource_group" "example" {
+  name     = var.azurerm_resource_group_name
+  location = var.location
+}
+
+# Create security group
+resource "azurerm_network_security_group" "example" {
+  name                = var.azurerm_network_security_group_name
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+}
+
+# Create a virtual network
+resource "azurerm_virtual_network" "example" {
+  name                = var.azurerm_virtual_network_name
+  resource_group_name = azurerm_resource_group.example.name
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.example.location
+}
+
+# Create a subnet
+resource "azurerm_subnet" "example" {
+  name                 = var.azurerm_subnet_name
+  resource_group_name  = azurerm_resource_group.example.name
+  virtual_network_name = azurerm_virtual_network.example.name
+  address_prefixes     = ["10.0.0.0/24"]
+
+  delegation {
+    name = "managedinstancedelegation"
+
+    service_delegation {
+      name = "Microsoft.Sql/managedInstances"
+      actions = [
+        "Microsoft.Network/virtualNetworks/subnets/join/action",
+        "Microsoft.Network/virtualNetworks/subnets/prepareNetworkPolicies/action",
+        "Microsoft.Network/virtualNetworks/subnets/unprepareNetworkPolicies/action"
+      ]
+    }
+  }
+}
+
+# Associate subnet and the security group
+resource "azurerm_subnet_network_security_group_association" "example" {
+  subnet_id                 = azurerm_subnet.example.id
+  network_security_group_id = azurerm_network_security_group.example.id
+}
+
+# Create a route table
+resource "azurerm_route_table" "example" {
+  name                          = "routetable-mi-terraform"
+  location                      = azurerm_resource_group.example.location
+  resource_group_name           = azurerm_resource_group.example.name
+  disable_bgp_route_propagation = false
+  depends_on = [
```

Comment:
Is this depends on necessary? We generally do not want to add explicit dependencies unless absolutely necessary.

Reply:
yes, you need to create this first

Reply:
@urosran  Why does the subnet need to exist before the route table is created? They should be able to be created in parallel, then the association below waits for both to exist before it connects them.

Reply:
TL;DR: To even associate a subnet with a route table you need one to exist therefore to make sure that the association will always be made and you won't have some failiure where route table is created but subent isn't you serialize them. This process is a fraction of the total time so we're not worried about it at all.

For more info check the docs https://docs.microsoft.com/en-us/azure/virtual-network/manage-route-table

Reply:
We do not like to use explicit depends_on statements unless necessary and in this case, unless I am issing something, it should no be necessary. Terraform creates implicit dependencies based on resource references so the association will create an implicit dependency to both route table and subnet resulting in the association only being made when both of those have completed successfully. See this dependency graph for clarification:
![graph-no-dependson](https://user-images.githubusercontent.com/2914336/195415685-da00a015-9588-4992-813f-ac44411958a4.svg)